### DISCUSSION TITLE
Issue with creating anonymous sharing link

### DISCUSSION THREAD

**Discussion by gautamdsheth** [pnp/pnpcore#1201] [Q&A] [answered]
Maybe I am doing this wrong, but when I am trying to create an anonymous link following the example in the docs, I get this error:

```
[parameters] Cannot convert the literal '2023-06-17T17.04.08Z' to the expected type 'Edm.DateTimeOffset'.
```

Sample code:

```
var shareLinkRequestOptions = new AnonymousLinkOptions()
{
    Type = ShareType.View,
    Password = "PnP Rocks!",
    ExpirationDateTime = DateTime.Now.AddDays(5),
};


var share = file.CreateAnonymousSharingLink(shareLinkRequestOptions);
```

Not sure if this is an error in PnP Core or I am doing this wrong.

**Comment by jansenbe** [comment]
@gautamdsheth : that's weird, I just ran our test case (https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core.Test/Security/PermissionTests.cs#L183-L235) that does exactly that and it works. Is the site collection in question configured for anon sharing? If you can do the same sharing from the UI then it is. 

Can you send me a Fiddler trace if the above does not help?

**Reply by gautamdsheth** [reply]
It works if I don't specify the expiration date. Maybe some datetime culture thing specific to my machine ?? Not sure, quite weird though
Sending you the fiddler trace.

**Reply by jansenbe** [reply, marked as answer]
https://github.com/pnp/pnpcore/commit/d00ff1c3ff2128bc57eba42c706ec1b2235a3b1a was pushed to fix this.
